didcomm/did_doc/did_resolver.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from didcomm.common.types import DID
+from didcomm.did_doc.did_doc import DIDDoc
+
+
+class DIDResolver(ABC):
+    """DID Resolver to resolver a DID to a DID DOC."""
+
+    @abstractmethod
+    async def resolve(self, did: DID) -> Optional[DIDDoc]:
```

Comment:
As I understand it, the ability to request a specific version of a DID Doc will be important for handling key rotations in DIDComm V2. I think we need a parameter to optionally capture requested version and pass it on to the resolvers.

Reply:
@dbluhm @ashcherbakov @dhh1128 It is quite interesting question. I think in current version DID Comm doesn't provide any approach or suggestions on how to handle versions. 

I suggest:

- Keep it as is
- Create a ticket to specification repo

Reply:
I fully agree that it's worth creation of a ticket, as it's not discussed in the DIDComm spec.

Actually I think we should add the field to DIDDoc resolver interface, it looks quite natural there.

Another question is how DIDComm spec (and hence DIDComm API) should reflect it.
For example, if I want to verify a signature from the past, how do I know what version of DID DOC should I use?
It looks like we may need to require the `versionID` be part of DID URL of signing keys...

Reply:
Actually, now that I think through this a bit more, specifying a given version of a DID Document is usually done by DID URL (@ashcherbakov beat me to that point in his last comment). So perhaps the correct answer here is to wholly delegate version concerns to the resolver and assume that information is sufficiently represented in the passed `DID`. Given that, I don't think my original suggestion makes sense.

Reply:
> It looks like we may need to require the versionID be part of DID URL of signing keys

I believe we should wait until it is reflected in specification and verify only using current version of DID Doc for now. 

Despite obvious value version gives for this concrete use case it opens outdated keys usage as serious attack vector. Just consider

* Alice has compromised key and Chuck knows it
* Alice rotates the key
* but Chuck uses keys he knows to produce fake message to Bob.

For Bob it becomes quite not obvious to trust this message or not.

Reply:
Yes. Let's create a ticket in DIDComm spec, and adjust the API is needed depending on the results of that discussion.